### ISSUE TITLE
`prefer-string-replace-all` not working with optional chaining

### ISSUE THREAD

**Issue by 2754github** [sindresorhus/eslint-plugin-unicorn#2769]
`prefer-string-replace-all` ignores `test1` but reports `test2`.
Is this the intended behavior, or should `test1` also be reported?

```typescript
const test1 = (Math.random() < 0.5 ? 'string' : null)?.replace(/\s/g, '');

const test2 = 'string'.replace(/\s/g, '');
```

**Comment by fisker** [comment]
Are you sure you are using the latest version? The first case should be reported.
